girder/utility/s3_assetstore_adapter.py

```diff
@@ -441,12 +443,46 @@ def stream():
                 if endByte is None or endByte > file['size']:
                     endByte = file['size']
                 headers = {'Range': 'bytes=%d-%d' % (offset, endByte - 1)}
+            else:
+                offset = 0
+                if endByte is None or endByte > file['size']:
+                    endByte = file['size']
+            # Our request can get interrupted for several reasons.  If it does
+            # we will typically get some form of IOError.  In this case, we
+            # want to retry it up to a point.
+            envval = os.environ.get('GIRDER_S3_DOWNLOAD_RETRIES')
+            maxRetriesWithoutData = int(envval) if str(envval).isdigit() else 3
 
             def stream():
-                pipe = requests.get(url, stream=True, headers=headers)
-                for chunk in pipe.iter_content(chunk_size=BUF_LEN):
-                    if chunk:
-                        yield chunk
+                streamOffset = offset
+                retries = 0
+                halt = False
+                while streamOffset < endByte and not halt:
+                    try:
+                        pipe = requests.get(url, stream=True, headers=headers)
+                        for chunk in pipe.iter_content(chunk_size=BUF_LEN):
+                            if chunk:
+                                streamOffset += len(chunk)
+                                try:
+                                    yield chunk
+                                    # If we actually got any data, reset our
+                                    # retry count
+                                    retries = 0
+                                except Exception:
+                                    # if the exception occurred because of the
+                                    # consumer, just stop
+                                    halt = True
+                                    raise
+                        halt = True
+                    except IOError as exc:
+                        retries += 1
+                        if halt or retries >= maxRetriesWithoutData:
+                            if not hasattr(exc, 'errno'):
```

Comment:
I would just add a comment to let future us know that this is to accommodate fusepy

Reply:
See [b15fc04](https://github.com/girder/girder/commit/b15fc040278d828ffda923b2fc0bd28e62674f95)